Singularity/Singularity/serialization/XSerializer.cs

```diff
@@ -0,0 +1,173 @@
+
+            //Write the xml-data of the object to the file
+            ser.WriteObject(xdwriter, toSerialize);
+
+            //Write xml-tail
+            xdwriter.WriteEndDocument();
+
+            xdwriter.Flush();
+            fs.Flush();
+            fs.Close();
+        }
+
+        /// <summary>
+        /// Deserialize the object of the given xml-file.
+        /// </summary>
+        /// <param name="filepath">The xml-file I was talking about</param>
+        /// <returns>A List containing all objects of the given file. Currently we should only have files with one object, though.</returns>
+        private static List<object> Deserialize(string filepath)
+        {
+            var ser = new NetDataContractSerializer();
+            var fs = new FileStream(filepath, FileMode.Open);
+            var reader = XmlDictionaryReader.CreateTextReader(fs, new XmlDictionaryReaderQuotas());
+            var deserializedObject = new List<object>();
+
+            while (reader.Read())
+            {
+                if (ser.IsStartObject(reader))
+                {
+                    var o = ser.ReadObject(reader);
+                    deserializedObject.Add(o);
+                }
+                break;
+            }
+            fs.Flush();
+            fs.Close();
+
+            return deserializedObject;
+        }
+
+        /// <summary>
+        /// Save the given object in a xml file with the given name. Already existing files with that name will be overridden.
+        /// The destination will be  %USERPROFILE%\Saved Games\Singularity (the directory will be created if it doesnt exist).
+        /// </summary>
+        /// <param name="toSave">The object that should be saved</param>
+        /// /// <param name="name">The name of the file. Don't forget to add .xml at the end!!!</param>
+        public static void Save(object toSave, string name)
+        {
+            var path = @"%USERPROFILE%\Saved Games\Singularity";
+            path = Environment.ExpandEnvironmentVariables(path);
+            if (!Directory.Exists(path))
+            {
+                Directory.CreateDirectory(path);
+            }
+            path = path + @"\" + name; 
+            Serialize(toSave, path);  
+        }
+
+        /// <summary>
+        /// Load the Xml-file containing whatever data. Remember that in our Saves there should only be one object, contact me if you want to change that.
+        /// It may be, that a typecast is needed since this method returns an object.
+        /// </summary>
+        /// <param name="path">The Xml-file I was talking about</param>
+        /// <returns>The Object that has been loaded</returns>
+        public static object Load(string path)
+        {
+            var loadedObjects = Deserialize(path);
+            if (loadedObjects.Count == 0)
+            {
+                throw new Exception("There are no deserialized Objects. Most likely your .xml file is empty.");
```

Comment:
Microsoft specifically states in their documentation to never throw a general Exception unless absolutely necessary. In this case I recommend using an IOException

Reply:
Oh I didnt know that. I guess I'll change that then.